cirq-google/cirq_google/transformers/target_gatesets/google_cz_gateset.py

```diff
@@ -0,0 +1,97 @@
+
+from typing import Any, Dict, List, Sequence, Type, Union
+import cirq
+
+
+class GoogleCZTargetGateset(cirq.CZTargetGateset):
+    """CZTargetGateset implementation tailored to Google devices."""
+
+    def __init__(
+        self,
+        atol: float = 1e-8,
+        eject_paulis: bool = False,
+        additional_gates: Sequence[Union[Type[cirq.Gate], cirq.Gate, cirq.GateFamily]] = (),
+    ):
+        """Initializes GoogleCZTargetGateset.
+
+        Args:
+            atol: A limit on the amount of absolute error introduced by the transformation.
+            eject_paulis: Whether to enable postprocess transformers `cirq.eject_z` and
+             `cirq.eject_phased_paulis`. Do not enable this if the circuit contains gates with
+             `cirq_google.PhysicalZTag` as the tag will be erased by these transformers. Defaults to
+             False.
+            additional_gates: Sequence of additional gates / gate families which should also
+              be "accepted" by this gateset. This is empty by default.
+        """
+        super().__init__(atol=atol, allow_partial_czs=False, additional_gates=additional_gates)
+        self.eject_paulis = eject_paulis
+        self._additional_gates_repr_str = ", ".join(
+            [cirq.ops.gateset._gate_str(g, repr) for g in additional_gates]
+        )
+
+    @property
+    def postprocess_transformers(self) -> List[cirq.TRANSFORMER]:
+        """List of transformers which should be run after decomposing individual operations.
+
+        If `eject_paulis` is enabled in the constructor, adds `cirq.eject_phased_paulis` and
+        `cirq.eject_z` in addition to postprocess_transformers already available in
+        `cirq.CompilationTargetGateset`.
+        """
+        transformers: List[cirq.TRANSFORMER] = [
+            cirq.create_transformer_with_kwargs(
+                cirq.merge_single_qubit_moments_to_phxz, atol=self.atol
+            ),
+            cirq.create_transformer_with_kwargs(cirq.drop_negligible_operations, atol=self.atol),
+            cirq.drop_empty_moments,
+        ]
+
+        if self.eject_paulis:
+            return (
+                transformers[:1]
+                + [
+                    cirq.create_transformer_with_kwargs(cirq.eject_phased_paulis, atol=self.atol),
+                    cirq.create_transformer_with_kwargs(cirq.eject_z, atol=self.atol),
+                ]
+                + transformers[1:]
+            )
+        return transformers
+
+    def __repr__(self) -> str:
+        return (
+            'cirq_google.GoogleCZTargetGateset('
+            f'atol={self.atol}, '
+            f'eject_paulis={self.eject_paulis}, '
+            f'additional_gates=[{self._additional_gates_repr_str}]'
+            ')'
+        )
+
+    def _value_equality_values_(self) -> Any:
+        return self.atol, self.eject_paulis, frozenset(self.additional_gates)
+
+    @classmethod
+    def _json_namespace_(cls) -> str:
+        return 'cirq.google'
```

Comment:
Shouldn't the json namespace be `cirq_google` instead of `cirq.google` ? Wasn't `cirq.google` a legacy namespace to support backwards compatibility before migration?